src/main/java/io/antmedia/rest/BroadcastRestService.java

```diff
@@ -342,6 +377,43 @@ public Result removeEndpoint(@ApiParam(value = "Broadcast id", required = true)
 		return result;
 	}
 
+	@ApiOperation(value = "Remove third pary rtmp end point from the stream. For the stream that is broadcasting, it will stop immediately", notes = "", response = Result.class)
+	@DELETE
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Path("/{id}/rtmp-endpoint")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Result removeEndpointV2(@ApiParam(value = "Broadcast id", required = true) @PathParam("id") String id, 
+			@ApiParam(value = "RTMP url of the endpoint that will be stopped.", required = true) @QueryParam("endpointServiceId") String endpointServiceId
+			){
+		
+		//Get rtmpURL with broadcast
+		String rtmpUrl = null;
+		
+		if(endpointServiceId != null && getDataStore().get(id) != null && !getDataStore().get(id).getEndPointList().isEmpty()) {
+			for(Endpoint endpoint: getDataStore().get(id).getEndPointList()) {
+				if(endpoint.getEndpointServiceId().equals(endpointServiceId)) {
+					rtmpUrl = endpoint.getRtmpUrl();
```

Comment:
why do you try to get then rtmpUrl?

Reply:
I get it. You probably don't want to change anything in datastore interface. I think it should change because it make software complex in this way.

Reply:
We need to detect RTMP URL for the getMuxAdaptor(id).stopRtmpStreaming(rtmpUrl). Please check 407. line.